requirements/doc/doc-requirements.txt

```diff
@@ -12,6 +12,7 @@ colorspacious
 ipython
 ipywidgets
 ipykernel
+networkx>=3.2.1
```

Comment:
I don't think we'd take `networkx` as a doc build dependency, unless there is something _very_ compelling here that couldn't be shown without it.  Any chance you can just put the data in as numpy vectors?

Reply:
It's not a matter of a data format. The example uses a couple of functions from NetworkX that are graph-specific and difficult to reinvent.

NetworkX is a natural choice for a graph-based example. I am unsure why using it in doc dependencies could be problematic.

Reply:
There are a few reasons. Our users need to be able to run our examples using our dependencies without needing to download an external library. Also because it's another dependency we need to load, slowing down each CI build. Finally, it can lead to issues of the libraries diverge.  

I don't know of any of our examples that use downstream libraries, otherwise we would have hundreds of examples from the many packages that use us as a drawing backend. 

Again if you would be willing to put the line segments into numpy, that could work.  Or perhaps networkx would like to have this example.